EngineLayer/MetaMorpheusEngine.cs

```diff
@@ -167,6 +167,59 @@ public static List<MatchedFragmentIon> MatchFragmentIons(Ms2ScanWithSpecificMass
             return matchedFragmentIons;
         }
 
+        //NGlycopeptide usually contain high charge Y ions. The purpose of this function is to try match all Y ion with different charges.
+        public static List<MatchedFragmentIon> MatchOriginFragmentIons(Ms2ScanWithSpecificMass scan, List<Product> theoreticalProducts, CommonParameters commonParameters)
```

Comment:
this seems to be mostly a duplicate of `MatchFragmentIons` - can you merge the two methods? maybe add a bool parameter set to false by default which looks for all the charge states of Y ions?

Reply:
This is an old solution for N-glycopeptides. It is not a good solution. I deleted it and will come up with a better one for N-glyco search update in the future. Thanks.